pydantic/_internal/_known_annotated_metadata.py

```diff
@@ -96,22 +96,6 @@
         CONSTRAINTS_TO_ALLOWED_SCHEMAS[c].update(schemas)
 
 
-def add_js_update_schema(s: cs.CoreSchema, f: Callable[[], dict[str, Any]]) -> None:
-    def update_js_schema(s: cs.CoreSchema, handler: GetJsonSchemaHandler) -> dict[str, Any]:
-        js_schema = handler(s)
-        js_schema.update(f())
-        return js_schema
-
-    if 'metadata' in s:
-        metadata = s['metadata']
-        if 'pydantic_js_functions' in s:
-            metadata['pydantic_js_functions'].append(update_js_schema)
-        else:
-            metadata['pydantic_js_functions'] = [update_js_schema]
-    else:
-        s['metadata'] = {'pydantic_js_functions': [update_js_schema]}
-
-
```

Comment:
Again... shouldn't be anywhere but `json_schema.py`, but this shouldn't require a functional wrapper here.